docs/docs/05-developer-guide/09-grpc/40-advanced/00-await-workflow-events.md

```diff
@@ -0,0 +1,74 @@
+3. `workflowEventsToIgnore` is a repeated field of IDs of `WorkflowEvents` that you want to ignore. This gives the client the ability to ignore `WorkflowEvent`s that have already been awaited. See [Ignoring WorkflowEvents](#ignoring-workflowevents) for an example.
+
+## Examples
+
+### Awaiting a single `WorkflowEvent`
+
+```java
+WorkflowEvent event = client.awaitWorkflowEvent(AwaitWorkflowEventRequest.newBuilder()
+            .setWfRunId(WfRunId.newBuilder().setId("your-workflow-run-id"))
+            .addEventDefIds(WorkflowEventDefId.newBuilder().setName("my-workflow-event-def"))
+            .build());
+```
+
+### Using Deadlines
+
+Upon the execution of a `THROW_EVENT` node, LittleHorse will always ensure that your `WorkflowEvent`s are thrown and returned to any clients awaiting them. However, you may still find it useful to set a [gRPC deadline](https://grpc.io/docs/guides/deadlines/) on your `AwaitWorkflowEvent` request in case a `WorkflowEvent` is not thrown within a specified period of time.
+
+```java
+Properties props = getConfigProps();
+LHConfig config = new LHConfig(props);
+LittleHorseBlockingStub client = config.getBlockingStub();
+
+WorkflowEvent event = client.withDeadlineAfter(1000, TimeUnit.MILLISECONDS)
+                .awaitWorkflowEvent(AwaitWorkflowEventRequest.newBuilder()
+                        .setWfRunId(WfRunId.newBuilder().setId("your-workflow-run-id"))
+                        .build());
+```
+
+:::info
+You can configure a gRPC deadline for any LittleHorse Client request, not just `AwaitWorkflowEvent`! If the request does not complete within the specified time, it will be automatically canceled and return gRPC Status Code 1 `CANCELLED`. 
+:::
+
+### Ignoring `WorkflowEvent`s
+
+Since a single `WfRun` may throw multiple `WorkflowEvent`s with the same `WorkflowEventDefId`, clients have the ability to "ignore" `WorkflowEvent`s that have already been awaited. Any `WorkflowEvent` specified within the `workflowEventsToIgnore` field will be ignored.
+
+```java
+// The first WorkflowEvent awaited by the client
+WorkflowEvent event1 = client.awaitWorkflowEvent(AwaitWorkflowEventRequest.newBuilder()
+            .setWfRunId(WfRunId.newBuilder().setId("your-workflow-run-id"))
+            .addEventDefIds(WorkflowEventDefId.newBuilder().setName("my-workflow-event-def"))
+            .build());
+
+// The second WorkflowEvent awaited by the client
+WorkflowEvent event2 = client.awaitWorkflowEvent(AwaitWorkflowEventRequest.newBuilder()
+    .setWfRunId(WfRunId.newBuilder().setId("your-workflow-run-id"))
+    .addEventDefIds(WorkflowEventDefId.newBuilder().setName("my-workflow-event-def"))
+    // Ignore any WorkflowEvents matching the first one received
+    // highlight-next-line
+    .addWorkflowEventsToIgnore(event1.getId())
+    .build());
+```
```

Comment:
Can you please add "Waiting for multiple `WorkflowEventDef`s," which is supposed to handle the case when a `WfRun` might throw one of several different event types?